scripts/self-hosted-kind/setup-host.yaml

```diff
@@ -0,0 +1,168 @@
+# MIT License
+#
+# Copyright (c) 2021 Mert Bora Alper and EASE lab
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# Execute this playbook using 
+#   ansible-playbook -u YOUR_SSH_USERNAME -i REMOTE_HOSTNAME, setup-host.yaml
+
+---
+- name: Setup Knative-on-KinD Host
+  hosts: all
+  tasks:
+    - name: Remove any old Docker installations
+      become: yes
+      apt:
+        state: absent
+        purge: yes
+        autoremove: yes
+        name:
+          - docker
+          - docker-engine
+          - docker.io
+          - containerd
+          - runc
+
+    - name: Remove Docker directories
+      become: yes
+      file:
+        state: absent
+        path: "{{ item }}"
+      loop:
+        - /var/lib/docker
+        - /var/lib/containerd
+
+    - name: Add custom repository keys
+      become: yes
+      ansible.builtin.apt_key:
+        url: "{{ item }}"
+      loop:
+        - https://download.docker.com/linux/ubuntu/gpg
+
+    - name: Add custom repositories
+      become: yes
+      ansible.builtin.apt_repository:
+        repo: "{{ item }}"
+        update_cache: no
+      loop:
+        - ppa:git-core/ppa
+        - deb [arch=amd64] https://download.docker.com/linux/ubuntu bionic stable
+
+    - name: Upgrade packages
+      become: yes
+      apt:
+        update_cache: yes
+        upgrade: safe
+
+    - name: Install required packages
```

Comment:
for docker, consider using their convenience script
https://docs.docker.com/engine/install/ubuntu/#install-using-the-convenience-script

Reply:
I have already written Ansible code for that, which I believe is more resilient to errors (being declarative in nature).